src/controller/auth.php

```diff
@@ -1175,3 +1183,262 @@
     //redirect
     $this->package('global')->redirect('/auth/login');
 });
+
+/**
+ * Process the SSO Login
+ *
+ * @param Request $request
+ * @param Response $response
+ */
+$cradle->get('/auth/sso/login/:service_type', function ($request, $response) {
```

Comment:
use `'/auth/sso/login/oauth1'` and `'/auth/sso/login/oauth2'` instead of `'/auth/sso/login/:service_type'`